### ISSUE TITLE
Sound device handover between different emulator windows does not work

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#267]
After replacing Musashi by Moira, it is now possible to run multiple emulator instances in parallel (with Musashi, only one instance could have been active at the same time, because there was only one core that had to be shared among instances). Moira is C++ and we can have as many instances as we want. In other words: It's now possible to rescue two princesses simultaneously. 

I've noticed that the sound device doesn't get released when the user switches back and forth between different emulator instances. I've opened up a new issue for that, so it won't be forgotten.

Other features are more important at the moment such as adding trace flag support. Since cputester supports trace flag testing, time feels right to add this functionality.

**Comment by dirkwhoffmann** [comment]
Embarrassing error in class `AudioEngine`🙄:

This ...

```
convenience init?(withPaula proxy: PaulaProxy) {
    
        self.init()
        paula = amigaProxy!.paula
```

... has to be this:

```
    convenience init?(withPaula proxy: PaulaProxy) {
    
        self.init()
        paula = proxy
```
